pkg/client/dtclient/settings_cache.go

```diff
@@ -0,0 +1,50 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dtclient
+
+type SettingsCache struct {
```

Comment:
could/should we make this a generic key-value cache? We'll need a schema and config API object cache soon as well

Reply:
@UnseenWizzard  it's coming :) For now just imagine to read `T` insead of `DownloadSettingsObject` :D